### PR TITLE
Improve conversation list scrolling performance

### PR DESCRIPTION
**Pull request by vytis**
### Issues

On older devices scrolling conversation list is quite choppy.

### Solutions

Multiple small issues were isolated:
* Setting various values on `UserImageView` and `AvatarImageView` would always redraw everything.   
     * **Solution:** redraw only if the value being set is different from old one.
* Creating empty attributed strings with helpers. 
     * **Solution:** Use `NSAttributedString()` instead.
* Posting a notification when conversation list scrolls. 
     * **Solution:** Don't post it and remove unused code that was listening to it.
* `prepareForReuse` resets all values and triggers a redraw. 
     * **Solution:** Remove the methods as we set all the values when dequeuing the cells anyway.

### Result

On iPhone 6 scrolling conversation list up and down when from ~24fps to ~30fps.

### CODE REVIEW

**Comment by cardassian-tailor** [comment]
is this related in any way to  #3156

**Comment by vytis** [comment]
@cardassian-tailor no not related. The issue described there is something that we are working on separately. This should improve situation when scrolling through conversations, but does not touch what happens inside a conversation or search.

File: Wire-iOS/Sources/UserInterface/Availability/AvailabilityStringBuilder.swift [hunk 1]
```diff
@@ -47,7 +47,7 @@
             }
         }
         
-        guard let textColor = color else { return "".attributedString }
+        guard let textColor = color else { return NSAttributedString() }
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Availability/AvailabilityStringBuilder.swift]: Maybe more make sense to change the return type of this method to `NSAttributedString?`. 3 callers of this method are use the return as NSAttributedString?.

Comment (vytis) [reply Wire-iOS/Sources/UserInterface/Availability/AvailabilityStringBuilder.swift]: Will take a look, thanks!

Comment (vytis) [reply Wire-iOS/Sources/UserInterface/Availability/AvailabilityStringBuilder.swift]: @billypchan updated!

**Pull request merged**